### PR TITLE
feat(permanents): python interface for Glynn repeated permanent calcu…

### PR DESCRIPTION
**Pull request by icnots**
…lation

The permanent is calculated based on the input and output states which mean the
multiplicity of the specific rows and columns (e.g. (1..1) and (1..1) mean the
classical permanent calculation). This means the sum of input and output states
has to be equal.

Added python function for calculating the permanents with input and output
states of a matrix. If there are higher than 1 inputs or outputs the calculator
optimizes the calculation based on them.

### CODE REVIEW

File: piquassoboost/sampling/permanent_calculators_wrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+
+    // create PIC version of the parameters
+    pic::matrix matrix_cpp = numpy2matrix(matrix_pyobj);
+    pic::PicState_int64 input_state = numpy2PicState_int64(input_state_pyobj);
+    pic::PicState_int64 output_state = numpy2PicState_int64(output_state_pyobj);
+  
+    input_state.print_matrix();
+    output_state.print_matrix();
+    matrix_cpp.print_matrix();
```
Comment (Kolarovszki) [inline piquassoboost/sampling/permanent_calculators_wrapper.cpp:86]: We don't need these prints.

File: piquassoboost/sampling/CMakeLists.txt [hunk 2]
```diff
@@ -49,15 +60,33 @@ target_include_directories(Boson_Sampling_Utilities_wrapper PUBLIC
...
 
 install(TARGETS Boson_Sampling_Utilities_wrapper LIBRARY DESTINATION piquassoboost/sampling)
+install(TARGETS permanent_calculators LIBRARY 
+         DESTINATION piquassoboost/gaussian)
+
+
+
+
+
+
```
Comment (Kolarovszki) [inline piquassoboost/sampling/CMakeLists.txt:91]: Delete these whitespaces!

File: piquassoboost/sampling/permanent_calculators_wrapper.cpp (hunk 1 above)
Comment (icnots) [reply piquassoboost/sampling/permanent_calculators_wrapper.cpp]: Deleted.

File: piquassoboost/sampling/CMakeLists.txt (hunk 2 above)
Comment (icnots) [reply piquassoboost/sampling/CMakeLists.txt]: Deleted.

**Review by Kolarovszki — APPROVED** [review]

**Pull request merged**